Source/SA/RFmx SpecAn/RFmxSpecAn.cs

```diff
@@ -44,33 +40,313 @@ public struct AutoLevelConfiguration
+            specAn.Acp.Configuration.ConfigureFft(selectorString, acpCommonConfig.FftWindow, acpCommonConfig.FftPadding);
+            specAn.Acp.Configuration.ConfigureRbwFilter(selectorString, acpCommonConfig.RbwAuto, acpCommonConfig.Rbw_Hz, acpCommonConfig.RbwFilterType);
+            specAn.Acp.Configuration.ConfigureSweepTime(selectorString, acpCommonConfig.SweepTimeAuto, acpCommonConfig.SweepTimeInterval_s);
+        }
+        public static void ConfigureAcpCarrierChannel(RFmxSpecAnMX specAn, AcpCarrierChannelConfiguration[] acpCarrierChannelConfiguration, string selectorString = "")
+        {
+            specAn.Acp.Configuration.ConfigureNumberOfCarriers(selectorString,acpCarrierChannelConfiguration.Length);
+            for (int i = 0; i < acpCarrierChannelConfiguration.Length; i++)
+            {
+                string carrierString = RFmxSpecAnMX.BuildCarrierString2(selectorString, i);
+                specAn.Acp.Configuration.ConfigureCarrierIntegrationBandwidth(carrierString, acpCarrierChannelConfiguration[i].IntegrationBandwidth_Hz);
+                specAn.Acp.Configuration.ConfigureCarrierRrcFilter(carrierString, acpCarrierChannelConfiguration[i].RrcFilterEnabled, acpCarrierChannelConfiguration[i].RrcAlpha);
+                specAn.Acp.Configuration.ConfigureCarrierMode(carrierString, acpCarrierChannelConfiguration[i].Mode);
+                specAn.Acp.Configuration.ConfigureCarrierFrequency(carrierString, acpCarrierChannelConfiguration[i].Frequency);
+            }
+        }
+        public static void ConfigureAcpOffsetChannel(RFmxSpecAnMX specAn, AcpOffsetChannelConfiguration[] acpOffsetChannelConfiguration, string selectorString = "")
+        {
+            specAn.Acp.Configuration.ConfigureNumberOfOffsets(selectorString,acpOffsetChannelConfiguration.Length);
+
+            for (int i = 0; i < acpOffsetChannelConfiguration.Length; i++)
+            {
+                string offsetString = RFmxSpecAnMX.BuildOffsetString2(selectorString, i);
+                specAn.Acp.Configuration.ConfigureOffsetIntegrationBandwidth(offsetString, acpOffsetChannelConfiguration[i].IntegrationBandwidth_Hz);
+                specAn.Acp.Configuration.ConfigureOffset(offsetString, acpOffsetChannelConfiguration[i].OffsetFrequency_Hz,acpOffsetChannelConfiguration[i].OffsetSideBand,acpOffsetChannelConfiguration[i].OffsetEnabled);
+                specAn.Acp.Configuration.ConfigureOffsetPowerReference(offsetString, acpOffsetChannelConfiguration[i].OffsetPowerReferenceCarrier,acpOffsetChannelConfiguration[i].OffsetPowerReferenceSpecificIndex);
+                specAn.Acp.Configuration.ConfigureOffsetRelativeAttenuation(offsetString, acpOffsetChannelConfiguration[i].RelativeAttenuation_dB);
+                specAn.Acp.Configuration.ConfigureOffsetRrcFilter(offsetString, acpOffsetChannelConfiguration[i].RrcFilterEnabled, acpOffsetChannelConfiguration[i].RrcFilterAlpha);
+            }
+        }

```

Comment:
can we move these into a single function? im thinking we need arrays of carrier channel configurations and offset channel configurations embedded within the AcpConfiguration struct

Reply:
I would rather suggest to keep them as separate parameters but get them called within a single method call. Minor argument: I am also thinking with respect to the LV integration where such a big cluster is not comfortable to handle on the front panel